packages/donor/src/components/AppointmentSelect/AppointmentSelect.tsx

```diff
@@ -0,0 +1,171 @@
+  isSwapAppointment: boolean;
+}
+
+export default function AppointmentSelect({
+  availableAppointments,
+  isFetching,
+  onSlotSelected,
+  tooCloseDonationPopupProps,
+  appointmentToHide,
+  isSwapAppointment,
+}: AppointmentSelectProps) {
+  const [selectedHospital, setSelectedHospital] = useState<Hospital | "">("");
+
+  const sortedDonationDays = React.useMemo(() => {
+    const appointmentFilter = (appointment: AvailableAppointment) => {
+      const isValidHospital =
+        appointment.hospital === selectedHospital || !selectedHospital;
+
+      let isHiddenAppointment = false;
+
+      if (appointmentToHide) {
+        const isHiddenTime =
+          appointment.donationStartTimeMillis ===
+          appointmentToHide!.donationStartTimeMillis;
+        const isHiddenHospital =
+          appointment.hospital === appointmentToHide!.hospital;
+        isHiddenAppointment = isHiddenTime && isHiddenHospital;
+      }
+      return isValidHospital && !isHiddenAppointment;
+    };
+    const filteredResults = availableAppointments.filter(appointmentFilter);
+    return groupDonationDays(filteredResults);
+  }, [availableAppointments, selectedHospital, appointmentToHide]);
+
+  return (
+    <>
+      <div className={styles.dropdownContainer}>
+        <Select
+          analyticsName="show_appointments_in_hospital"
+          label={"הצג תורים ב:"}
+          className={styles.dropdown}
+          options={HospitalUtils.getHospitalOptions(
+            HospitalUtils.activeHospitals,
+            "הכל"
+          )}
+          value={selectedHospital}
+          onChange={setSelectedHospital}
+          isDisabled={isFetching}
+        />
+      </div>
+
+      <Popup
+        name="TooCloseAppointment"
+        title={"מועד קרוב מידי"}
+        buttonApproveText={"כן, אשר תור"}
+        open={tooCloseDonationPopupProps.open}
+        goBackText={"התחרטתי, חזרה למסך התורים"}
+        onApproved={tooCloseDonationPopupProps.onApproved}
+        onBack={tooCloseDonationPopupProps.onBack}
+        onClose={tooCloseDonationPopupProps.onClose}
+        image={WarningLogo}
+      >
+        טרם חלפו חודש ימים מיום תרומתך האחרון. האם את/ה בטוח/ה שברצונך{" "}
+        {isSwapAppointment ? "להחליף" : "לקבוע"} תור זה?
+      </Popup>
+
+      <div className={styles.donationsCard}>
+        {Donations(
+          selectedHospital,
+          isFetching,
+          sortedDonationDays,
+          onSlotSelected
+        )}
+      </div>
+    </>
+  );
+}
+
+function Donations(
+  selectedHospital: Hospital | "",
+  isFetching: boolean,
+  donationDays: DonationDay[],
+  onSlotSelected: (donationSlot: DonationSlotToBook) => void
+) {
+  if (isFetching) {
+    return (
+      <div className={styles.loading}>
+        <Player
+          autoplay
+          loop
+          src={DropAnimation}
+          className={styles.dropAnimation}
+        />
+        <div className={styles.loadingText}>מחפש תורים קרובים...</div>
+      </div>
```

Comment:
This should be a standalone component (used here and in other places).

Reply:
This is something I moved over to the refactor PR.
Should I return it?